generators/fusesoc

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import yaml
+import subprocess
+
+templ = """/*
+:name: {0}
+:description: {1}
+:files: {2}
+:incdirs: {3}
+:top_module: {4}
+:tags: {5}
+:timeout: {6}
+*/
+"""
+
+try:
+    third_party_dir = os.environ['THIRD_PARTY_DIR']
+    tests_dir = os.environ['TESTS_DIR']
+    conf_dir = os.environ['CONF_DIR']
+except KeyError:
+    print("Export the THIRD_PARTY_DIR, TESTS_DIR and CONF_DIR variables first")
+    sys.exit(1)
+
+try:
+    tests_subdir = sys.argv[1]
+except IndexError:
+    print("Usage: ./generator <subdir>")
+    sys.exit(1)
+
+fusesoc_conf_files = os.listdir(
+    os.path.abspath(os.path.join(conf_dir, "fusesoc-configs")))
+
+for fusesoc_conf_file in fusesoc_conf_files:
+    fusesoc_conf_file = os.path.abspath(
+        os.path.join(conf_dir, "fusesoc-configs", fusesoc_conf_file))
+
+    if not os.path.isfile(fusesoc_conf_file):
+        continue
+
+    with open(fusesoc_conf_file, 'r') as stream:
+        try:
+            yml_conf = yaml.safe_load(stream)
+        except yaml.YAMLError as err:
+            print(err)
+            sys.exit(1)
+
+    try:
+        yml_conf["name"]
+        yml_conf["description"]
+        yml_conf["top_module"]
+        yml_conf["tags"]
+        yml_conf["path"]
+        yml_conf["command"]
+        yml_conf["conf_file"]
+        yml_conf["test_file"]
+        yml_conf["timeout"]
+    except KeyError as err:
+        print("No key", err, "in file", fusesoc_conf_file)
+        continue
```

Comment:
What I meant is that you could wrap the whole code in try/except so you don't have to have a list of the required keys.
If you want to have a list of the required keys you could have used the `in` keyword and a regular `if` statement.

But let's not spend more time on it. This should work just fine.